src/site/content/en/blog/sms-otp-form/index.md

```diff
@@ -0,0 +1,256 @@
+    * `type="text"`
+    * `inputmode="numeric"`
+    * `autocomplete="one-time-code"`
+* Use `@${bound-domain} #${OTP}` as the last line of the SMS text message.
+* Use the Web OTP API.
+
+## Use the `<input>` element
+
+Using a form with an `<input>` element is the most important best practice you
+can follow as it will definitely work on any browser. Even if other items in
+this article don't work, the user can still enter and submit the OTP by doing it
+manually.
+
+```html
+<form action="/verify-otp" method="POST">
+  <input type="text"
+         inputmode="numeric"
+         autocomplete="one-time-code"
+         pattern="\d{6}"
+         required>
+</form>
+```
+
+The following are a few ideas to ensure an input field gets the best out of
+browser functionality.
+
+{% Aside 'gotchas' %}
+There are more general form best practices other than ones described below. [Sam
+Dutton](https://twitter.com/sw12)'s [Sign-in form best
+practices](https://web.dev/sign-in-form-best-practices/) is a great starting
+point.
+{% endAside %}
+
+### `type="text"`
+
+Because OTPs are usually 5 or 6 digit numbers, you may be tempted to use
+`type="number"` for an input field because it turns the mobile keyboard to be
+numbers only. However, it's not the best idea, because the browser expects an
+input field to be a countable number rather than a sequence of multiple numbers,
+and this causes unexpected behaviors. Using `type="number"` causes up and down
+buttons to be displayed beside the input field, and pressing these buttons
+increments or decrements the number and may remove preceding zeros.
+
+Use `type="text"` instead. This won't turn the mobile keyboard into numbers
+only, but that is fine because the next tip for using `inputmode="numeric"` does
+that job.
+
+### `inputmode="numeric"`
+
+Some websites use `type="tel"` for an input field which is smart because it
+turns the mobile keyboard to be number only (including `*` and `#`) when
+focused. Developers choose this  because, in the past, some browsers didn't
+support
+[`inputmode="numeric"`](https://developer.mozilla.org/docs/Web/HTML/Global_attributes/inputmode).
+But since [Firefox started supporting
+`inputmode="numeric"`](https://github.com/mdn/browser-compat-data/pull/6782),
+you no longer need to stick with the hack of using `type="tel"`. You can instead
+use `inputmode="numeric"` to turn the mobile keyboard into the numbers only one
+which is semantically better.
+
+### `autocomplete="one-time-code"`
+
+[`autocomplete="one-time-code"`](https://developer.mozilla.org/docs/Web/HTML/Attributes/autocomplete)
+works only on Safari 12 and later on iOS, iPadOS and macOS, but we strongly
+recommend using it, because this is an easy way to improve the SMS OTP
+experience on those platforms. Whenever a user receives an SMS text message with
+the form, the operating system will parse the OTP in the SMS heuristically and
+the keyboard will suggest that the user enter the OTP.
```

Comment:
```suggestion
[`autocomplete`](https://developer.mozilla.org/docs/Web/HTML/Attributes/autocomplete) 
attribute lets developers specify what permission the browser 
has to provide autocomplete assistance and informs the browser about the 
type of information expected in the field.

With `autocomplete="one-time-code"` whenever a user receives an SMS message while a 
form is open, the operating system will parse the OTP in the SMS heuristically and
the keyboard will suggest the OTP for the user to enter. It works only on Safari 12 and 
later on iOS, iPadOS, and macOS, but we strongly recommend using it, because it is an 
easy way to improve the SMS OTP experience on those platforms.
```